uiowa/uiowa
# [ITS] sprint 224 updates.

- [x] - Next 7 days text on https://its.prod.drupal.uiowa.edu/alerts
- [x] - All day check box on alert template
- [ ] - We want to train web editors on how to add an update to the alert using Drupal. We believe that the way things are set up now, the automatic email to helpdesk-notify@uiowa.edu is only triggered upon the creation of a new alert. Is it possible to make the automatic email get triggered by a new update to an existing alert?
- [x] - I was able to replicate the legend (https://its.prod.drupal.uiowa.edu/alerts) using a bulleted list with link titles to create the hovered text effect. I don't have access to the icons for the new alerts page, though.
- [x] - Can we allow editor roles to publish changes to service templates? Currently they cannot create or delete service pages, which is what we want. But if they have edits to an already established/published service, we'd like for the to be able to edit and publish instead of submitting their edits to review. This is similar to what we have with info centers and the third tier role on our current D7 site.
- [x] - Can we allow the editor role to replicate alerts. It looks like only publisher and webmaster role can replicate alerts and we'd like for editors to be able to as well. 






# How to test

We have a meeting with ITS to discuss the above unchecked items, so we do not need to test them right now. If you can think of any way to fix the issues that they describe before then, please comment with your suggestions.

1. Pull and install this branch
2. sync its and log in
3. go to `/alerts` and check that there is `next 7 days` text for both of the bottom blocks.
4. Edit the layout and delete the manually constructed Alert types legend block
5. Add the new custom `Alert types legend` block and observe that it displays all alert types and icons.
6. Go to an alert page an verify that, for the dates affected field, there is no All-day option anymore.
7. Masquerade as an editor and verify that you can publish unpublished services.



# Post deploy

- Place legend blocks on /alerts and /alerts/calendar
